src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -79,4 +82,15 @@ public void storeMessage(Message message) {
 
     return messages;
   }
+
+  public Set<String> getUsers(){
+    Set<String> users = new HashSet<>();
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    for(Entity entity : results.asIterable()) {
```

Comment:
Style nit: add a space after `for` (the space missing on the guided project site is a typo)